### DISCUSSION TITLE
Return errors with graphql-shield (nexus.js) in a resolver.

### DISCUSSION THREAD

**Discussion by trufa** [prisma/prisma#7582] [Q&A] [answered]
Hi all!

In [graphql-shield](https://github.com/maticzav/graphql-shield) it states:

To return custom error messages to your client, you can return error instead of throwing it.

The problem is that when using nexus.js with typescript, the resolver doesn’t allow to return an error.

Other than returning:

`return new Error("Custom error") as any
`

What can I do if I have something like this:

```
export default mutationField('signup', {
  type: AuthPayload, // Here should be OR ERROR
  async resolve(_parent, args, context: Context) {
```

Thank you very much in advance!

**Comment by ryands17** [comment, marked as answer]
@trufa 👋 

You can use union types in Nexus as shown in this [example](https://blog.logrocket.com/handling-graphql-errors-like-a-champ-with-unions-and-interfaces/).

**Reply by ryands17** [reply]
Let me know if that's not straightforward so that I can send you an example :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
